x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/droppable/on_drop_handler.ts

```diff
@@ -109,45 +110,27 @@ function onMoveCompatible(
 ) {
   const layer = state.layers[layerId];
   const sourceColumn = layer.columns[droppedItem.columnId];
-
-  const newColumns = {
-    ...layer.columns,
-    [columnId]: { ...sourceColumn },
-  };
-  if (shouldDeleteSource) {
-    delete newColumns[droppedItem.columnId];
-  }
-
-  const newColumnOrder = [...layer.columnOrder];
+  const indexPattern = state.indexPatterns[layer.indexPatternId];
+  let modifiedLayer = layer;
 
   if (shouldDeleteSource) {
-    const sourceIndex = newColumnOrder.findIndex((c) => c === droppedItem.columnId);
-    const targetIndex = newColumnOrder.findIndex((c) => c === columnId);
-
-    if (targetIndex === -1) {
-      // for newly created columns, remove the old entry and add the last one to the end
-      newColumnOrder.splice(sourceIndex, 1);
-      newColumnOrder.push(columnId);
-    } else {
-      // for drop to replace, reuse the same index
-      newColumnOrder[sourceIndex] = columnId;
-    }
-  } else {
-    // put a new bucketed dimension just in front of the metric dimensions, a metric dimension in the back of the array
-    // then reorder based on dimension groups if necessary
-    const insertionIndex = sourceColumn.isBucketed
-      ? newColumnOrder.findIndex((id) => !newColumns[id].isBucketed)
-      : newColumnOrder.length;
-    newColumnOrder.splice(insertionIndex, 0, columnId);
+    modifiedLayer = deleteColumn({
```

Comment:
Isn't this deleting the source column and all referenced columns from `modifiedLayer` which will be needed in `copyColumn`? Looks like it's safer to copy first and delete afterwards.